python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -899,8 +899,15 @@ def _fetch_existing_runs(
         return {}
 
     # fetch runs from the DB with only the run key tag
-    # note: while possible to filter more at DB level with tags - it is avoided here due to observed perf problems
-    runs_with_run_keys = instance.get_runs(filters=RunsFilter(tags={RUN_KEY_TAG: run_keys}))
+    # note: while possible to filter more at DB level with tags - it is avoided here due to observed
+    # perf problems
+    runs_with_run_keys = []
+    for run_key in run_keys:
```

Comment:
@prha These are not in a set but I assume we never have duplicates in here? that might change the behavior slightly.

Reply:
it doesn't really matter, since all we do is check for existence to prevent re-submitting.